packages/eslint-plugin/src/rules/no-confusing-void-expression.ts

```diff
@@ -167,6 +184,26 @@ export default createRule<Options, MessageId>({
         if (invalidAncestor.type === AST_NODE_TYPES.ReturnStatement) {
           // handle return statement
 
+          if (options.ignoreVoidInVoid) {
+            if (
+              invalidAncestor.parent.parent?.type ===
```

Comment:
[bug] We can't assume that `return` statement is always in the root-level block scope of the function. It may be inside blocks, if-else statements, switch statements, loops, etc.

So the following example is false positive (the rule shouldn't complain):

```ts
declare function returnsVoid(): void

function test(): void {
  {
    return returnsVoid();
  }
}
```

([playground](https://deploy-preview-8632--typescript-eslint.netlify.app/play/#ts=5.3.3&fileType=.ts&code=CYUwxgNghgTiAEAzArgOzAFwJYHtXzg2RlQGcA1HLYACgEoAueANyuACh2V1s94MQpDPSatq8AN7t4k6TIIgiJBUrKVq9ANxyAvux1A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1rI6YDNYyZgHNaANw6UAJvQAexaCiG90YANrhsORNGgdokADSatWLdkiVhnRQDVJUgJJN701fjiIT2AL7eAupp%2BPkA&tsconfig=&tokens=false))

Reply:
it seems like a case I hadn’t thought of. Thank you.